### PR TITLE
Fixes thermals

### LINKED ISSUE #28445: Thermals are broken
(WEB REPORT BY: spacedgod REMOTE: 206.221.180.138:7777)
# Revision
0d9909fbd717c95aae7d60ff0374d476cdb5b35b

# Description
Thermals don't show people in darkness or through walls. They're basically shitty mesons now.
# Steps to Reproduce
Wear thermals and look at someone in the dark. See nothing
# What you Expected
working thermal wallhacks
# What Actually Happened
shit coders broke because they suck

### PR DESCRIPTION
**Pull request by JTGSZ**
closes #28445

Untitled

Lets people with thermals see people through walls again, also cleans up the glasses.dm file and organizes it a bit.

### CODE REVIEW

File: code/modules/mob/hearing/virtualhearer.dm [hunk 1]
```diff
@@ -71,12 +71,12 @@ var/list/stationary_hearers = list(	/obj/item/device/radio/intercom,
 		sight = copying
 	if(adding)
 		sight |= adding
-		if(adding & SEE_TURFS)
+		if((adding & SEE_TURFS) || (adding & SEE_MOBS))
```
Comment (Exxion) [inline code/modules/mob/hearing/virtualhearer.dm]:
```suggestion
		if(adding & (SEE_TURFS | SEE_MOBS))
```
Bitwise checks can be combined like this

File: code/modules/mob/hearing/virtualhearer.dm [hunk 2]
```diff
@@ -71,12 +71,12 @@ var/list/stationary_hearers = list(	/obj/item/device/radio/intercom,
...
 			sight &= ~SEE_BLACKNESS
 	if(removing)
 		sight &= ~removing
-		if(removing & SEE_TURFS)
-			sight |= SEE_BLACKNESS
+		if((removing & SEE_TURFS) || (removing & SEE_MOBS))
```
Comment (Exxion) [inline code/modules/mob/hearing/virtualhearer.dm]:
```suggestion
		if(removing & (SEE_TURFS | SEE_MOBS))
```
Same here

File: code/modules/mob/hearing/virtualhearer.dm [hunk 3]
```diff
@@ -71,12 +71,12 @@ var/list/stationary_hearers = list(	/obj/item/device/radio/intercom,
...
+			sight |= SEE_BLACKNESS
```
Comment (Exxion) [inline code/modules/mob/hearing/virtualhearer.dm]:
```suggestion
			sight |= SEE_BLACKNESS
```
This was just some weird trailing whitespace your text editor didn't remove for some reason

File: code/modules/mob/hearing/virtualhearer.dm (hunk 1 above)
Comment (JTGSZ) [reply code/modules/mob/hearing/virtualhearer.dm]: cool, i didn't know that and I made the changes

**Pull request merged**